File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        for (int i : nums) {
           int count = hashMap.getOrDefault(i, 0) + 1;

            // If the count exceeds nums.length / 2, return the element
            if (count > nums.length / 2) {
                return i;
            }

            // Update the count in the HashMap
            hashMap.put(i, count);
        }
        return -1;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # Moore Algo it seems
        count = 0
        candidate = -1

        for i in nums:
            if count == 0:
                candidate = i
                count = 1
            elif candidate == i:
                count+=1
            else:
                count-=1
        
        return candidate